### ISSUE TITLE
Cypress | Docker and cypress included 6.1.0, complains about missing plugins

### ISSUE THREAD

**Issue by kezzakyle** [cypress-io/cypress#14281]
From following this helpful link: https://www.cypress.io/blog/2019/05/02/run-cypress-with-a-single-docker-command/

And mass googling (a few links as example):

https://github.com/cypress-io/cypress-docker-images/issues/265

https://github.com/cypress-io/cypress-docker-images/issues/178

I've managed to get the tests running correctly without plugin/support (by setting them to false in cypress config). BUT when I want to use cypress-axe to test accessibility and use docker, I receive the following.

The command to execute:

```
docker run -it -v $PWD:/e2e -w /e2e cypress/included:6.1.0 --spec "**/components/button.spec.js"
```

The resulting error:

```
Error: Cannot find module 'cypress-axe' from '/e2e/tests/UI/cypress/support'
This occurred while Cypress was compiling and bundling your test code. This is usually caused by:
- A missing file or dependency
- A syntax error in the file or one of its dependencies
Fix the error in your code and re-run your tests.
```

Also dependent on what version of the cypress included I use also get another error (I believe this error is coming from .babelrc, again missing plugin issue):

```
Error: Webpack Compilation Error
./tests/UI/cypress/integration/components/button.spec.js
Module build failed (from /root/.cache/Cypress/6.2.0/Cypress/resources/app/packages/server/node_modules/babel-loader/lib/index.js):
Error: Cannot find module '@babel/plugin-syntax-dynamic-import'
```

So, by the looks of the errors it cannot find any installed plugin dependencies as the dependencies are not inside the image/container as of yet. Which makes sense. I was wondering is there a way to leverage the cypress-axe plugin all from this one command? Something like:
```
docker run -it -v $PWD:/e2e -w /e2e cypress/included:6.1.0 /bin/bash -c 'npm install cypress-axe && cypress run --spec "**/components/button.spec.js"'
```
BUT wanting to avoid creating a docker file and just use one command to run cypress inside docker and use plugins. Along with caching this as we do not want to install the plugins each time we run the command.

Any suggestions would be much appreciated!

**Comment by jennifer-shehane** [comment]
If I understand you correctly, you've set `supportFile: false` in your Cypress config, but in order to use the cypress-axe plugin, you must import it into the `supportFile` via its instructions: https://www.npmjs.com/package/cypress-axe So, you need to specify the supportFile and import the plugin.

Let me know if I've misunderstood this.

**Comment by kezzakyle** [comment]
My config looks like this:

  "fixturesFolder": "cypress/fixtures",
  "integrationFolder": "cypress/integration",
  "numTestsKeptInMemory": 10,
  "pluginsFile": "cypress/plugins/index.js",
  "screenshotsFolder": "cypress/screenshots",
  "supportFile": "cypress/support/index.js",
  "testFiles": "**/*.*",
  "video": false,
  "videosFolder": "cypress/videos"
  
  Another suggestion I had was to create a docker layer above this one and install all required plugins but failed at that too.

**Comment by bahmutov** [comment]
I have added `cypress-axe` plugin in https://github.com/bahmutov/demo-docker-cypress-included#cypress-plugins and am testing it on CircleCI successfully. Please see what is different about your project from our example. For us to proceed in debugging this would require a reproducible example.

**Comment by kezzakyle** [comment]
Thanks for the prompt response @bahmutov! I managed to get it working too within your repo and crossed out all my theories of why this was not working. (babel.rc error theories and config theories) I ended up finding that on our staging server, node modules are not there, which is the cause of the issue and we do not want to npm i within our staging server. Is there anyway to run a cypress plugin via command line? Like how we are running cypress include with docker?
eg something like:

`docker run -it -v $PWD:/e2e -w /e2e cypress/included:6.2.1 cypress/magicPluginIncludeHere --spec "**/components/button.spec.js"`

**Comment by bahmutov** [comment]
Honestly, I do not see a workaround here - you need node_modules for the plugins to even be there, and they need to be installed. I would talk to your dev ops people on how you can deploy the staging server and maybe even Cypress tests separately, but run `npm install` before the Cypress tests.
